examples/features/interceptor/README.md

```diff
@@ -17,102 +17,104 @@ go run client/main.go
+In gRPC, interceptors can be categorized into two kinds based on the type of RPC
+calls they intercept. The first one is the **unary interceptor**, which
+intercepts unary RPC calls. The second is the **stream interceptor** which deals
+with streaming RPC calls. See the
+[gRPC docs](https://grpc.io/docs/guides/concepts.html#rpc-life-cycle) for an
+explanation about unary and streaming RPCs. Both the client and the server have
+their own types of unary and stream interceptors. Thus, there are four different
+types of interceptors in total.
 
 ### Client-side
 
 #### Unary Interceptor
 
-[`UnaryClientInterceptor`](https://godoc.org/google.golang.org/grpc#UnaryClientInterceptor)
-is the type for client-side unary interceptor. It is essentially a function type
-with signature: `func(ctx context.Context, method string, req, reply
-interface{}, cc *ClientConn, invoker UnaryInvoker, opts ...CallOption) error`.
-An implementation of a unary interceptor can usually be divided into three
-parts: pre-processing, invoking RPC method, and post-processing.
+The type for client-side unary interceptors is
+[`UnaryClientInterceptor`](https://godoc.org/google.golang.org/grpc#UnaryClientInterceptor).
+It is essentially a function type with signature: `func(ctx context.Context,
+method string, req, reply interface{}, cc *ClientConn, invoker UnaryInvoker,
+opts ...CallOption) error`. Unary interceptor implementations can usually be
+divided into three parts: pre-processing, invoking the RPC method, and
+post-processing.
 
 For pre-processing, users can get info about the current RPC call by examining
-the args passed in, such as RPC context, method string, request to be sent, and
-CallOptions configured. With the info, users can even modify the RPC call. For
-instance, in the example, we examine the list of CallOptions and see if call
-credential has been configured. If not, configure it to use oauth2 with token
-"some-secret-token" as fallback. In our example, we intentionally omit
-configuring the per RPC credential to resort to fallback.
-
-After pre-processing is done, use can invoke the RPC call by calling the
-`invoker`.
-
-Once the invoker returns the reply and error, user can do post-processing of the
-RPC call. Usually, it's about dealing with the returned reply and error. In the
+the args passed in. The args include the RPC context, method string, request to
+be sent, and the CallOptions configured. With this info, users can even modify
+the RPC call. For instance, in the example, we examine the list of CallOptions
+and check if the call credentials have been configured. If not, the interceptor
+configures the RPC call to use oauth2 with a token "some-secret-token" as a
+fallback. In our example, we intentionally omit configuring the per RPC
+credential to resort to the fallback.
+
+After pre-processing, users can invoke the RPC call by calling the `invoker`.
+
+Once the invoker returns a reply and error, users can do post-processing of the
```

Comment:
Remove "a reply and error"?

"do post-processing of" -> "post-process"

Reply:
Changes made.